toolchains/osx.toolchain.cmake

```diff
@@ -39,17 +39,40 @@ set( TNUN_os_suffix OSX )
 
 set( TNUN_cpu_archs default )
 
+set( TNUN_ABI x64 )
+
 add_compile_options( -mmmx -mfpmath=sse -mcx16 )
 
-#...mrmlj...todo...
-#if ( <arch> STREQUAL sse3 )
-#    set( XCODE_ATTRIBUTE_CFLAGS_i386   "-msse3  -march=prescott -mtune=core2"  )
-#    set( XCODE_ATTRIBUTE_CFLAGS_x86_64 "-mssse3 -march=core2    -mtune=corei7" )
-#elseif ( <arch> STREQUAL sse4.1 )
-#    set( XCODE_ATTRIBUTE_CFLAGS_i386   "-msse4.1 -march=core2 -mtune=core2"  )
-#    set( XCODE_ATTRIBUTE_CFLAGS_x86_64 "-msse4.1 -march=core2 -mtune=corei7" )
-#endif()
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
```

Comment:
this whole block can only go in if it is enabled only for generators other than xcode (as it makes no sense for xcode)
+ the MACOSX_CPP_LIBRARY* variables can be named more generic like TNUN_CPP_LIB
+ the clang+stdc++ and gcc+libc++ combinations and their permutations (if we really need them) can be made more generic (you can have the same situation on linux or even windows)...